lib/src/git.rs

```diff
@@ -1276,9 +1276,38 @@ pub fn reset_head(mut_repo: &mut MutableRepo, wc_commit: &Commit) -> Result<(),
             // Just overwrite if unborn (or conflict), which is also unusual.
             gix::refs::transaction::PreviousValue::MustExist
         };
-        let new_oid = new_head_target
-            .as_normal()
-            .map(|id| gix::ObjectId::from_bytes_or_panic(id.as_bytes()));
+
+        let new_oid = new_head_target.as_normal().map(|id| {
+            let single_parent_bookmark = wc_commit
+                .single_parent()
+                .and_then(|_| {
+                    mut_repo
+                        .view()
+                        .bookmarks()
+                        .filter(|(_, bookmark)| bookmark.local_target == &new_head_target)
```

Comment:
This is a scan over all bookmarks. The commit template uses an `CommitRefsIndex` to accelerate this, but as far as I can tell there's nothing to improve this available here.

Reply:
Maybe you can use `local_bookmarks_for_commit()`? It's not optimized in any sense, though.